### PR TITLE
Use --no-deps when installing compiled requirements files

### PR DESCRIPTION
**Pull request by relud**
to work around https://github.com/pypa/pip/issues/9644

see also https://github.com/mozilla/gcp-ingestion/pull/2001

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -5,10 +5,8 @@ click==8.0.4
 gcsfs==2022.2.0
 gcloud==0.18.3
 gitpython==3.1.27
-google-api-core==1.31.5  # transitive dep that needs dependabot updates
 google-cloud-bigquery==2.34.0
 google-cloud-storage==2.1.0
-googleapis-common-protos==1.54.0  # transitive dep that needs dependabot updates
```
Comment (relud) [inline requirements.in:11]: remove previous workaround which was having no impact

**Review by sean-rose — CHANGES_REQUESTED** [review]
You read my mind! (thanks for tackling this)

- There are some `pip install` commands in `.circleci/config.yml` which also need to be updated.
- A similar change needs to be made wherever pip-sync is used (e.g. `--pip-args "--no-deps"`), or pip-sync could be swapped out for `pip install`.
  - On a related note, in the readme's "Installing bqetl" section, running pip-sync in step 4 ends up uninstalling bqetl from the venv because bqetl isn't in the requirements files, which would be a reason to use `pip install` instead.

**Comment by relud** [comment]
> You read my mind! (thanks for tackling this)
> 
>     * There are some `pip install` commands in `.circleci/config.yml` which also need to be updated.

👍 

>     * A similar change needs to be made wherever pip-sync is used (e.g. `--pip-args "--no-deps"`), or pip-sync could be swapped out for `pip install`.

CI caches the pip venv, and needs to use `pip-sync` to uninstall removed dependencies to ensure reproducibility.

>       * On a related note, in the readme's "Installing bqetl" section, running pip-sync in step 4 ends up uninstalling bqetl from the venv because bqetl isn't in the requirements files, which would be a reason to use `pip install` instead.

not using pip-sync is outside the scope of this PR

**Comment by relud** [comment]
> > ```
> >   * On a related note, in the readme's "Installing bqetl" section, running pip-sync in step 4 ends up uninstalling bqetl from the venv because bqetl isn't in the requirements files, which would be a reason to use `pip install` instead.
> > ```
> 
> not using pip-sync is outside the scope of this PR

but we can retain bqetl with `<(echo mozilla-bigquery-etl)`, which i've added.

File: .circleci/config.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ jobs:
           command: |
             python3.8 -m venv venv/
             venv/bin/pip install pip-tools --constraint requirements.in
-            venv/bin/pip-sync
+            venv/bin/pip-sync --pip-args=--no-deps
```
Comment (sean-rose) [inline .circleci/config.yml:37]: By default I think pip-sync only looks for `requirements.txt`, so should this specify both `requirements.txt` and `java-requirements.txt` like is done elsewhere?
```suggestion
            venv/bin/pip-sync --pip-args=--no-deps requirements.txt \
              java-requirements.txt
```

File: docs/cookbooks/testing.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ python3.8 -m venv venv/
 ./venv/bin/pip install pip-tools -c requirements.in
 
 # install python dependencies with pip-sync (provided by pip-tools)
-./venv/bin/pip-sync
+./venv/bin/pip-sync --pip-args=--no-deps
```
Comment (sean-rose) [inline docs/cookbooks/testing.md:13]: Similarly:
```suggestion
./venv/bin/pip-sync --pip-args=--no-deps requirements.txt java-requirements.txt
```

**Review by sean-rose — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (relud) [reply .circleci/config.yml:37]: java requirements are intentionally left out here, as they are optional and should not be required for unit tests.

File: docs/cookbooks/testing.md (hunk 3 above)
Comment (relud) [reply docs/cookbooks/testing.md:13]: yes, this spot should include java requirements, because it runs integration tests below

**Pull request merged**